### PR TITLE
Add CreateSymbolicLink

### PR DESCRIPTION
**Pull request by godmyoh**
This PR adds definitions of `CreateSymbolicLink` into `winbase.d`.

There are many missing functions in `winbase.d`. But I couldn't port them in my free time...

### CODE REVIEW

**Comment by CyberShadow** [comment]
FWIW, it's possible to create symbolic links without this function. This function seems to have been added in Windows Vista, but symbolic links were an NTFS feature for a while before that.

https://github.com/CyberShadow/ae/blob/fd8b46feff780d1fe977534882fd4b359c232f1c/sys/file.d#L1241-L1341

File: src/core/sys/windows/winbase.d [hunk 1]
```diff
@@ -1589,6 +1589,14 @@ static if (_WIN32_WINNT >= 0x410) {
     alias DWORD EXECUTION_STATE;
 }
 
+// CreateSymbolicLink
+static if (_WIN32_WINNT >= 0x600) {
+    enum {
+        SYMBOLIC_LINK_FLAG_DIRECTORY = 0x1,
```
[the remark this answers is not in the document]
Comment (godmyoh) [reply src/core/sys/windows/winbase.d]: My code was based on older version of `winbase.h`...
Latest `winbase.h` has `SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE` and has no `VALID_SYMBOLIC_LINK_FLAGS`.
So I fixed them based on latest ones.

**Comment by godmyoh** [comment]
> FWIW, it's possible to create symbolic links without this function. This function seems to have been added in Windows Vista, but symbolic links were an NTFS feature for a while before that.
> 
> https://github.com/CyberShadow/ae/blob/fd8b46feff780d1fe977534882fd4b359c232f1c/sys/file.d#L1241-L1341

Thank you for your information.
I'll check `DeviceIoControl` and `ioapiset.h`.

**Comment by ZombineDev** [comment]
@CyberShadow Thanks, very interesting. I guess the main benefit of the the dedicated function (compared to the your low-level approach) is that since one of the updates in Windows 10 you don't need admin privileges to use it.

**Pull request merged**